src/Bridge/Laravel/DbToolsServiceProvider.php

```diff
@@ -0,0 +1,270 @@
+
+namespace MakinaCorpus\DbToolsBundle\Bridge\Laravel;
+
+use Illuminate\Contracts\Config\Repository;
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Foundation\Console\AboutCommand;
+use Illuminate\Support\ServiceProvider;
+use MakinaCorpus\DbToolsBundle\Anonymization\AnonymizatorFactory;
+use MakinaCorpus\DbToolsBundle\Anonymization\Anonymizer\AnonymizerRegistry;
+use MakinaCorpus\DbToolsBundle\Anonymization\Config\Loader\ArrayLoader;
+use MakinaCorpus\DbToolsBundle\Anonymization\Config\Loader\PhpFileLoader;
+use MakinaCorpus\DbToolsBundle\Anonymization\Config\Loader\YamlLoader;
+use MakinaCorpus\DbToolsBundle\Bridge\Symfony\DependencyInjection\DbToolsConfiguration;
+use MakinaCorpus\DbToolsBundle\Command\Anonymization\AnonymizeCommand;
+use MakinaCorpus\DbToolsBundle\Command\Anonymization\AnonymizerListCommand;
+use MakinaCorpus\DbToolsBundle\Command\Anonymization\CleanCommand;
+use MakinaCorpus\DbToolsBundle\Command\Anonymization\ConfigDumpCommand;
+use MakinaCorpus\DbToolsBundle\Command\BackupCommand;
+use MakinaCorpus\DbToolsBundle\Command\CheckCommand;
+use MakinaCorpus\DbToolsBundle\Command\RestoreCommand;
+use MakinaCorpus\DbToolsBundle\Command\StatsCommand;
+use MakinaCorpus\DbToolsBundle\Configuration\Configuration;
+use MakinaCorpus\DbToolsBundle\Configuration\ConfigurationRegistry;
+use MakinaCorpus\DbToolsBundle\Database\DatabaseSessionRegistry;
+use MakinaCorpus\DbToolsBundle\Error\ConfigurationException;
+use MakinaCorpus\DbToolsBundle\Storage\DefaultFilenameStrategy;
+use MakinaCorpus\DbToolsBundle\Storage\FilenameStrategyInterface;
+use MakinaCorpus\DbToolsBundle\Storage\Storage;
+use Symfony\Component\Config\Definition\Processor;
+
+class DbToolsServiceProvider extends ServiceProvider
+{
+    public function register(): void
+    {
+        $this->app->extend('config', function (Repository $config, Application $app) {
+            $dbToolsConfig = $config->get('db-tools', []);
+            $dbToolsConfig = ['db_tools' => $dbToolsConfig];
+
+            $processor = new Processor();
+            $configuration = new DbToolsConfiguration(false, true);
+
+            $dbToolsConfig = $processor->processConfiguration($configuration, $dbToolsConfig);
+            $dbToolsConfig = DbToolsConfiguration::fixLegacyOptions($dbToolsConfig);
+            $dbToolsConfig = DbToolsConfiguration::appendPostConfig($dbToolsConfig);
+
+            $config->set('db-tools', $dbToolsConfig);
+
+            return $config;
+        });
+
+        $this->app->singleton(DatabaseSessionRegistry::class, function (Application $app) {
+            return new LaravelDatabaseSessionRegistry($app->make('db'));
+        });
+
+        $this->app->singleton(ConfigurationRegistry::class, function (Application $app) {
+            /** @var Repository $config */
+            $config = $app->make('config');
+
+            $defaultConfig = new Configuration(
+                backupBinary: $config->get('db-tools.backup_binary', '/usr/bin/pg_dump'),
```

Comment:
You should not set any default in configuration, the PHP classes will do their job on their own.

Reply:
This is true for all variables, except for the `storage_directory` one.

Reply:
Apparently `storage_directory` is not the only exception. For example, `Configuration` class has no default value for the backup binary, nor the backup options. Should I complete this class?

Reply:
You dont need defaults for binaires, each backupper and restorer class provides its own.

Reply:
Ok! I understand better now.